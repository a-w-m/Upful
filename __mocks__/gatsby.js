const React = require("react")
const gatsby = jest.requireActual("gatsby")

const mockDate = new Date("2019-04-07T10:20:30Z")

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      to,
      ...rest
    }) =>
      React.createElement("a", {
        ...rest,
        href: to,
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn().mockReturnValue({
        "allMarkdownRemark": {
          "edges": [
            {
              "node": {
                "fields": {
                  "slug": "/about/"
                },
                "frontmatter": {
                  "title": "About"
                }
              }
            },
            {
              "node": {
                "fields": {
                  "slug": "/returns/"
                },
                "frontmatter": {
                  "title": "Returns"
                }
              }
            }
          ]
        },
        "site": {
          "siteMetadata": {
            "menuLinks": {
              "categories": [
                {
                  "name": "women",
                  "slug": "/women/"
                },
                {
                  "name": "kids",
                  "slug": "/kids/"
                }
              ]
            }
          }
        },
        "file": {
          "id": "f1fc4c65-65c9-5962-b6a8-9a265bb2e617",
          "childImageSharp": {
            "gatsbyImageData": {
              "layout": "constrained",
              "backgroundColor": "#f8f8f8",
              "images": {
                "fallback": {
                  "src": "/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png",
                  "srcSet": "/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w",
                  "sizes": "(min-width: 250px) 250px, 100vw"
                },
                "sources": [
                  {
                    "srcSet": "/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,\n/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w",
                    "type": "image/webp",
                    "sizes": "(min-width: 250px) 250px, 100vw"
                  }
                ]
              },
              "width": 250,
              "height": 250
            }
          }
        }
  }),
  Date: jest.spyOn(global.Date, "now").mockImplementation(() => mockDate),
}
