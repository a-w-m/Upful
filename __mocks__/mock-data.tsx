export const pageQueryData = {
  data: {
    markdownRemark: {
      frontmatter: {
        customField1: {
          name: "size",
          values: [
            {
              name: "small",
              priceChange: 0,
            },
            {
              name: "medium",
              priceChange: 0,
            },
            {
              name: "large",
              priceChange: 0,
            },
          ],
        },
        customField2: {
          name: "color",
          values: [
            {
              name: "blue",
              priceChange: 0,
            },
            {
              name: "grey",
              priceChange: 0,
            },
            {
              name: "black",
              priceChange: 0,
            },
          ],
        },
        title: "Pietsie Atlin Blouse",
        price: 240,
        date: "2020-12-29T00:00:00.000Z",
        description:
          "Pietsie Campbell creates all of her garments out of deadstock, vintage fabrics, and natural fibers. Each collection is made in sunny California and embodies the effortlessly chic look we all strive for. Our favorite loose, flowing blouse in a cotton and linen blend. Slightly fitted across the shoulders. Wear tucked or loose, with the sleeves rolled up or down. Delicate pleating at shoulders, bell sleeves, pleated cuffs. Trapeze shape, easy fit. Pulls over, vintage shell buttons on a placket that ends mid way. Delicate hand-pleated detail on chest, stand-up prairie collar.",
        id: "pietsie-atlin-blouse",
        image: {
          childImageSharp: {
            fluid: {
              base64:
                "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q==",
              aspectRatio: 1,
              src:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg",
              srcSet:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/f836f/pietsie-atlin-blouse.jpg 200w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/2244e/pietsie-atlin-blouse.jpg 400w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg 800w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/16310/pietsie-atlin-blouse.jpg 1024w",
              sizes: "(max-width: 800px) 100vw, 800px",
            },
          },
        },
      },
      fields: {
        slug: "/pietsie-atlin-blouse/",
      },
    },
  },
}

export const pageQueryDataNullCustomFields = {
  data: {
    markdownRemark: {
      frontmatter: {
        customField1: null,
        customField2: null,
        title: "Pietsie Atlin Blouse",
        price: 240,
        date: "2020-12-29T00:00:00.000Z",
        description:
          "Pietsie Campbell creates all of her garments out of deadstock, vintage fabrics, and natural fibers. Each collection is made in sunny California and embodies the effortlessly chic look we all strive for. Our favorite loose, flowing blouse in a cotton and linen blend. Slightly fitted across the shoulders. Wear tucked or loose, with the sleeves rolled up or down. Delicate pleating at shoulders, bell sleeves, pleated cuffs. Trapeze shape, easy fit. Pulls over, vintage shell buttons on a placket that ends mid way. Delicate hand-pleated detail on chest, stand-up prairie collar.",
        id: "pietsie-atlin-blouse",
        image: {
          childImageSharp: {
            fluid: {
              base64:
                "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q==",
              aspectRatio: 1,
              src:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg",
              srcSet:
                "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/f836f/pietsie-atlin-blouse.jpg 200w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/2244e/pietsie-atlin-blouse.jpg 400w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg 800w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/16310/pietsie-atlin-blouse.jpg 1024w",
              sizes: "(max-width: 800px) 100vw, 800px",
            },
          },
        },
      },
      fields: {
        slug: "/pietsie-atlin-blouse/",
      },
    },
  },
}

export const fluid = {
  base64:
    "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q==",
  aspectRatio: 1,
  src:
    "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg",
  srcSet:
    "/static/184dc14fc5e96a12c4f7c6ee0f71f32a/f836f/pietsie-atlin-blouse.jpg 200w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/2244e/pietsie-atlin-blouse.jpg 400w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/14b42/pietsie-atlin-blouse.jpg 800w,\n/static/184dc14fc5e96a12c4f7c6ee0f71f32a/16310/pietsie-atlin-blouse.jpg 1024w",
  sizes: "(max-width: 800px) 100vw, 800px",
}


