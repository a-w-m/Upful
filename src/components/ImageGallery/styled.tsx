import styled from "styled-components"
import device from "../layout/mediaQuery"
export const Wrapper = styled.div`
  width: auto;
  height: auto;
  display: flex;
  flex-direction: column;
  row-gap: 1rem;

  @media ${device.laptop}{
    grid-area: gallery;
    justify-self: center;
    align-self: flex-start;
    
  }
`
export const UL = styled.ul`
  width: 100%;
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  margin: 0;
  gap: 1.4rem;

  & > li {
    list-style: none;
    flex: 0 0 6.8rem;
    margin-bottom: 0;
  }
`
export const LI = styled.li``
