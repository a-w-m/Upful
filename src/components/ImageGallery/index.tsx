import React, { ChangeEvent } from 'react'
import {GatsbyImage} from 'gatsby-plugin-image'
import {P} from "../interfaces/index"
import {Wrapper, UL, LI} from "./styled"

interface Props{
    images: P.ImageNode[]  
    dispatch: P.Dispatch
}
const ImageGallery:React.FC<Props> =(props)=>{
    const {images, dispatch} = props

    const handleClick = (img: P.Image['childImageSharp']['gatsbyImageData'])=>{
        dispatch({type: "image", payload: img})
    }

if (images.length <= 1){
    return null
}

    return (
        <Wrapper>
            <UL>
            {images.map((image, index)=>{
                return (
                    <LI key = {index} onClick = {()=>{handleClick(image.node.childImageSharp.gatsbyImageData)}}>
                    <GatsbyImage image = {image.node.childImageSharp.gatsbyImageData} alt = "" ></GatsbyImage>
                    </LI>
                )
            
            })}
            </UL>

        </Wrapper>
    )

}

export default ImageGallery