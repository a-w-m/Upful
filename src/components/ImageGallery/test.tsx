import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import ImageGallery from "./index"
import { imageGalleryData } from "../../../__mocks__/mock-data"

describe("ImageGallery Component", () => {
  const mockDispatch = jest.fn()

  test("matches snapshot", () => {
    const { container } = render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("dispatch is called when gallery image is clicked ", () => {
    render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    const image = screen.getAllByRole("listitem")[0]
    fireEvent.click(image)
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})

test("no list rendered when no images supplied", () => {
  const { container } = render(
    <ImageGallery images={[]} dispatch={jest.fn()}></ImageGallery>
  )
  expect(container.firstChild).toBeNull()
})
