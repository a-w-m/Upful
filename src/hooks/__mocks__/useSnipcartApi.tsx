import { getSnipcartApiData } from "../../utils/test/data"

const useSnipcartApi =  jest.fn().mockImplementation(() => {
    return [getSnipcartApiData().state, getSnipcartApiData().dispatch]
  })

export default useSnipcartApi
