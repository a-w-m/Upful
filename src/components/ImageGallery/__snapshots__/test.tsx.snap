// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGallery Component matches snapshot 1`] = `
<div
  class="sc-bdfBwQ bOsYoh"
>
  <ul
    class="sc-gsTCUz jxwfsg"
  >
    <li
      class="sc-dlfnbm"
    >
      <div
        class="gatsby-image-wrapper"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden;"
      >
        <div
          aria-hidden="true"
          style="padding-top: 100%;"
        />
        <img
          alt=""
          aria-hidden="true"
          data-placeholder-image=""
          decoding="async"
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q=="
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <picture>
          <source
            srcset="/static/184dc14fc5e96a12c4f7c6ee0f71f32a/c65bc/pietsie-atlin-blouse.webp 150w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/078c3/pietsie-atlin-blouse.webp 300w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/6d09e/pietsie-atlin-blouse.webp 600w"
            type="image/webp"
          />
          <img
            alt=""
            data-gatsby-image-ssr=""
            data-main-image=""
            data-src="/static/184dc14fc5e96a12c4f7c6ee0f71f32a/a89ca/pietsie-atlin-blouse.jpg"
            data-srcset="/static/184dc14fc5e96a12c4f7c6ee0f71f32a/96deb/pietsie-atlin-blouse.jpg 150w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/0fdf4/pietsie-atlin-blouse.jpg 300w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/a89ca/pietsie-atlin-blouse.jpg 600w"
            decoding="async"
            loading="lazy"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
          />
        </picture>
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t)e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset")),e.complete&&(e.style.opacity=1)}
        </script>
      </div>
    </li>
    <li
      class="sc-dlfnbm"
    >
      <div
        class="gatsby-image-wrapper"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden;"
      >
        <div
          aria-hidden="true"
          style="padding-top: 100%;"
        />
        <img
          alt=""
          aria-hidden="true"
          data-placeholder-image=""
          decoding="async"
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAQDBQL/xAAVAQEBAAAAAAAAAAAAAAAAAAACAf/aAAwDAQACEAMQAAABo5/qM3ZaNxlHOsAv/8QAGxAAAgMAAwAAAAAAAAAAAAAAAhEAAQMQMjP/2gAIAQEAAQUCNAZG+Nms1Z1Ow5etKf/EABcRAQADAAAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8BYmv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAgEBPwEh/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEQITER/9oACAEBAAY/AmU6WQxdixGH/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFRQXGB/9oACAEBAAE/IXc+4GAexFvZzo0GrKj8QZgMPIs6gqnJhwyf/9oADAMBAAIAAwAAABBg6AP/xAAYEQACAwAAAAAAAAAAAAAAAAABEBExQf/aAAgBAwEBPxCjCYX/xAAZEQACAwEAAAAAAAAAAAAAAAAAARARITH/2gAIAQIBAT8QWqzThx//xAAaEAEAAwEBAQAAAAAAAAAAAAABABEhMUFR/9oACAEBAAE/EHjoQXw9Zt3EBTGMhU2ZF4YEW35sIxq0nx8nAaNWAyzXA74cHfsrTdJZ2Gw7kOz/2Q=="
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <picture>
          <source
            srcset="/static/184dc14fc5e96a12c4f7c6ee0f71f32a/c65bc/pietsie-atlin-blouse.webp 150w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/078c3/pietsie-atlin-blouse.webp 300w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/6d09e/pietsie-atlin-blouse.webp 600w"
            type="image/webp"
          />
          <img
            alt=""
            data-gatsby-image-ssr=""
            data-main-image=""
            data-src="/static/184dc14fc5e96a12c4f7c6ee0f71f32a/a89ca/pietsie-atlin-blouse.jpg"
            data-srcset="/static/184dc14fc5e96a12c4f7c6ee0f71f32a/96deb/pietsie-atlin-blouse.jpg 150w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/0fdf4/pietsie-atlin-blouse.jpg 300w,/static/184dc14fc5e96a12c4f7c6ee0f71f32a/a89ca/pietsie-atlin-blouse.jpg 600w"
            decoding="async"
            loading="lazy"
            style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
          />
        </picture>
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t)e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset")),e.complete&&(e.style.opacity=1)}
        </script>
      </div>
    </li>
  </ul>
</div>
`;
